File: Number System/Primilarity Test/Primality_Test.py
def solve():
    n = int(input())  # Read the input number n
    if n == 1:  # Special case: 1 is not a prime number
        print('no')
        return
    if n == 2:  # Special case: 2 is the smallest prime number
        print('yes')
        return
    i = 2  # Start checking from the smallest prime number
    while i * i <= n:  # Only check divisors up to the square root of n
        if n % i == 0:  # If n is divisible by i, it's not a prime number
            print('no')
            return
        i += 1  # Increment i to check the next potential divisor
    print('yes')  # If no divisors found, n is a prime number

t = 1  # Number of test cases
t = int(input())  # Read the number of test cases
for i in range(t):  # Loop over each test case
    solve()  # Call the solve function for each test case

File: Number System/Primilarity Test/Primality_Test.cpp
#include <bits/stdc++.h>
using namespace std;

#define endl '\n'
#define FOR(i, a, b) for(int i = (a); i < (b); i++)
#define FORk(i, a, b, k) for(int i = (a); i < (b); i += k)
#define RFOR(i, a, b) for(int i = (a); i >= (b); i--)
#define RFORk(i, a, b, k) for(int i = (a); i >= (b); i -= k)

void solve() {
    int n;
    cin >> n; // Read the integer n

    // Check if n is 1
    if (n == 1) {
        cout << "no" << endl; // 1 is not a prime number
        return;
    }

    // Check if n is 2
    if (n == 2) {
        cout << "yes" << endl; // 2 is a prime number
        return;
    }

    // Loop from 2 to sqrt(n) to check for factors
    for(int i = 2; i * i <= n; i++) {
        if (n % i == 0) {
            cout << "no" << endl; // If n is divisible by i, it's not a prime number
            return;
        }
    }

    // If no factors are found, n is a prime number
    cout << "yes" << endl;
}

int main() {
    ios_base::sync_with_stdio(false); // Disable synchronization with C I/O for faster execution
    cin.tie(NULL); // Untie cin from cout
    cout.tie(NULL); // Untie cout from cin

    int t;
    cin >> t; // Read the number of test cases

    while (t--) {
        solve(); // Solve each test case
    }

    return 0;
}
